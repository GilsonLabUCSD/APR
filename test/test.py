import os
import sys
import shutil
import glob as glob
import subprocess as sp

# This script tests the error_estimate module with a set of pre-generated restraints.dat files.

targetValUpper = -5.70
targetValLower = -5.71

# Copy over the scripts to the test folder
print('Copying over scripts ...')
for file in glob.glob('../*.py'):
    if file != '../test.py':   # Do not copy file that has the same name with the current script!
        shutil.copy2(file, './')

for file in glob.glob('../apr.in'):
    shutil.copy2(file, './')

for file in glob.glob('test.out'):
    os.remove(file)

print ('Testing apr analysis ... Sit tight :)')
sp.call('python2 apr.py analysis -i apr.in > test.out', shell = True)

# Grab the final result of binding free energy from the output file
with open('test.out') as fr:
    lines = fr.read().splitlines()
    lines = list(line for line in lines if line)

for i, line in enumerate(lines):
    line = lines[i].split()
    if line[0] == 'Binding':
        val = float(line[3])
        break

if (targetValLower < val < targetValUpper):
    print ('Test passed!')

# Clean up
for file in glob.glob('*.py'):
    if file != 'test.py':   # Do not delete itself!
        os.remove(file)

for file in glob.glob('*.pyc'):
    os.remove(file)

for file in glob.glob('apr.in'):
    os.remove(file)

for file in glob.glob('*.dat'):
    os.remove(file)
