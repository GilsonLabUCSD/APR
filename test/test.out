**********************************************************************************
 Welcome to APR: a tool for binding calculations. Version 1.1
 Written by:
   Niel M. Henriksen, Jian (Jane) Yin, David R. Slochower
 and the project leader:
   Michael K. Gilson 
 Copyright (c) 2016-2017, University of California, San Diego
**********************************************************************************
The current APR scripts may not be directly applied to protein systems, especially
for those with buried binding sites. Careful adjustments of the protocols and scripts
will be needed, based on the requirements of every particular system.

Checking the version of Python installed ...
You are using Python 2.7, which is perfect for running the APR scripts.

Analyzing the binding calculations ...
all in kcal/mol                   Mean                SEM
Attachment:                     4.072694            0.048592
Translation (pulling):          8.775702            0.221298
Release-standard:              -7.141322               N/A
Binding free energy:           -5.707073            0.226570
